contracts/discovery/GNSStorage.sol

```diff
@@ -18,17 +23,31 @@ contract GNSV1Storage is Managed {
     // Bonding curve formula
     address public bondingCurve;
 
-    // graphAccountID => subgraphNumber => subgraphDeploymentID
-    // subgraphNumber = A number associated to a graph accounts deployed subgraph. This
-    //                  is used to point to a subgraphID (graphAccountID + subgraphNumber)
-    mapping(address => mapping(uint256 => bytes32)) public subgraphs;
+    // [DEPRECATED] Stores what subgraph deployment a particular subgraph targets
+    // A subgraph is defined by (graphAccountID, subgraphNumber)
+    // A subgraph can target one subgraph deployment (bytes32 hash)
+    // (graphAccountID, subgraphNumber) => subgraphDeploymentID
+    // TODO: delete this mapping when NFT claimed on old subgraphs
+    mapping(address => mapping(uint256 => bytes32)) public __DEPRECATED_subgraphs;
+
+    // Every time an account creates a subgraph it increases a per-account sequence ID
+    // account => seqID
+    mapping(address => uint256) public nextAccountSeqID;
+
+    // Stores all the signal deposited on a subgraph
+    // (graphAccountID, subgraphNumber) => SubgraphData
+    mapping(address => mapping(uint256 => IGNS.SubgraphData)) public nameSignals;
+
+    // [DEPRECATED] ERC-1056 contract reference
+    // This contract is used for managing identities
+    IEthereumDIDRegistry private __DEPRECATED_erc1056Registry;
 
-    // graphAccountID => subgraph deployment counter
-    mapping(address => uint256) public graphAccountSubgraphNumbers;
+    // -- (v1.1) Storage --
 
-    // graphAccountID => subgraphNumber => NameCurationPool
-    mapping(address => mapping(uint256 => IGNS.NameCurationPool)) public nameSignals;
+    // TODO: use it whenever an "old" subgraph NFT was claimed to maintain compatibility
```

Comment:
I think this strategy will work for upgrading permissionlessly. 

We could also consider that we just mint the NFTs on migration for our users. We could do it automatically, like prefill this data, or we could make it more permissionless by making it a function anyone can call. 

The benefits are that we do not have to explain to the users how to do the migrate. I think a bit about the LEND --> AAVE migration. It is something that had to be explained, communicated, and had some split for a while on the two tokens. And then,  even governance votes to close the migration contract, many months in the future.

But just an idea. it can be discussed

Reply:
Yes, I was thinking on making the claim function for the NFT public. Then to get it called for all existing subgraphs after an upgrade. That would initialize the required structures so anyone can call the contract functions with the new primary key.

Reply:
i like it